Improve pytest plugin documentation (#65631)
zephyrproject-rtos/zephyr: doc/develop/test/twister.rst

        Only one fixture can be defined per testcase and the fixture name has to
        be unique across all tests in the test suite.

.. _pytest_root:

    pytest_root: <list of pytest testpaths> (default pytest)
        Specify a list of pytest directories, files or subtests that need to be
        executed when a test case begins to run. The default pytest directory is
        ``pytest``. After the pytest run is finished, Twister will check if

>         the tst case passed or failed according to the pytest report.

Reviewer: ```suggestion
        the test case passed or failed according to the pytest report.
```

>         Specify a list of pytest directories, files or subtests that need to be
>         executed when test case begin to running, default pytest directory is
>         pytest. After pytest finished, twister will check if this case pass or
>         fail according to the pytest report. List of exemplary and valid
>         pytest roots is presented below:

Reviewer: ```suggestion
        Specify a list of pytest directories, files or subtests that need to be
        executed when a test case begins to run. The default pytest directory is
        ```pytest``. After the pytest run is finished, Twister will check if 
        the tst case passed or failed according to the pytest report.
        As an example, a list of valid pytest roots is presented below:
```

        .. code-block:: yaml

          harness_config:
            pytest_root:
              - "pytest/test_shell_help.py"
              - "../shell/pytest/test_shell.py"
              - "/tmp/test_shell.py"
              - "~/tmp/test_shell.py"
              - "$ZEPHYR_BASE/samples/subsys/testsuite/pytest/shell/pytest/test_shell.py"
              - "pytest/test_shell_help.py::test_shell2_sample"  # select pytest subtest

>               - "pytest/test_shell_help.py::test_shell2_sample[param_a]"  # select pytest parametrized subtest

Reviewer: looks like this might not be correctly indented (needs an extra 1-character positive indend), not sure if this causes rendering issues (the CI build expired) but prob worth fixing anyway

Author: I added extra indend - it is rendered properly (now and before too)
